docs/components/VersionSelector.tsx

```diff
@@ -49,35 +50,42 @@ const STYLES_SELECT_ELEMENT = css`
 `;
 
 type Props = {
-  version: string;
-  onSetVersion: (value: string) => void;
   style?: React.CSSProperties;
 };
 
-const VersionSelector: React.FC<Props> = ({ version, style, onSetVersion }) => (
-  <div css={STYLES_SELECT} style={style}>
-    <label css={STYLES_SELECT_TEXT} htmlFor="version-menu">
-      <div>{Utilities.getUserFacingVersionString(version, LATEST_VERSION, BETA_VERSION)}</div>
-      <ChevronDownIcon style={{ height: '16px', width: '16px' }} />
-    </label>
-    {
-      // hidden links to help test-links spidering
-      VERSIONS.map(v => (
-        <a key={v} style={{ display: 'none' }} href={`/versions/${v}/`} />
-      ))
-    }
-    <select
-      id="version-menu"
-      css={STYLES_SELECT_ELEMENT}
-      value={version}
-      onChange={e => onSetVersion(e.target.value)}>
-      {VERSIONS.map(v => (
-        <option key={v} value={v}>
-          {Utilities.getUserFacingVersionString(v, LATEST_VERSION, BETA_VERSION)}
-        </option>
-      ))}
-    </select>
-  </div>
-);
+const VersionSelector: React.FC<Props> = ({ style }) => {
+  const { version, hasVersion, setVersion } = usePageApiVersion();
+
+  // Don't render the version selector on non-versioned pages
+  if (!hasVersion) {
+    return null;
+  }
+
+  return (
+    <div css={STYLES_SELECT} style={style}>
+      <label css={STYLES_SELECT_TEXT} htmlFor="version-menu">
+        <div>{Utilities.getUserFacingVersionString(version, LATEST_VERSION, BETA_VERSION)}</div>
+        <ChevronDownIcon style={{ height: '16px', width: '16px' }} />
+      </label>
+      {
+        // hidden links to help test-links spidering
+        VERSIONS.map(v => (
```

Comment:
```suggestion
        VERSIONS.map(version => (
```
I much prefer to write out the variable name for readability